### ISSUE TITLE
Broken handling of current stats

### ISSUE THREAD

**Issue by tvm1** [CleverRaven/Cataclysm-DDA#12473]
I've had a little session with gdb today and found out, that all these neat CBMs actually don't do what they seem to do.

All code that is accessing player public members (int_cur, str_cur, etc.) and using these for calculations (smashing, melee, reading, etc.) is using nerfed values without CBM bonus.

Pieces of code that actually grab the stats via creature::get_int() (that actually handles the bonus correctly) work fine.

This also applies to stuff like meth which buff your stats, but don't get ever used in calculations.

This relates closely to #12390. 

Obviously, there are several solutions to this problem.

On the long run, it would be best if there was universal and standardized way of accessing player stats to avoid future problems.

**Comment by Coolthulhu** [comment]
`int_cur` etc. should probably be made private and only accessed using getters.
Changes to them get overwritten really quickly anyway.

**Comment by i2amroy** [comment]
>int_cur etc. should probably be made private and only accessed using getters.

I believe that was the eventual intention, the stuff using ::get_int() is the stuff that has been rewritten, but it was so big of a change at the start that the whole thing wasn't done at once (and is thus causing the problems we are seeing now).

**Comment by Coolthulhu** [comment]
OK, I scanned the code and it's as I suspected: the order is wrong.

`player::reset_stats()` calls `Character::reset_stats()` which calls `Creature::reset_stats()` which sets all the `str_cur`, `dex_cur` etc. to values equal to `[stat]_max + [stat]_bonus`.

This alone isn't a problem. However the function `reset_stats()` is generally invoked as a part of `reset()`, which calls `reset_bonuses()` and does exactly what it name says it does.

As a result, the bonuses for stats are first zeroed, then used (as zeros), then recalculated.

Just swapping the order of the two functions invoked in `reset()` should help, though it might also have some other effects.
